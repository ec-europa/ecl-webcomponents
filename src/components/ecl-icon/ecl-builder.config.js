const path = require('path');
const pkg = require('./package.json');

const nodeModules = path.resolve(__dirname, '../../../node_modules');
const outputFolder = './build';

// SCSS includePaths
const includePaths = [nodeModules];

const banner = `${pkg.name} - ${pkg.version} Built on ${new Date().toISOString()}`;

module.exports = {
  copy: [
    {
      from: [path.resolve(nodeModules, '@ecl/resources-social-media-icons/dist/sprites/icons-social-media.svg')],
      to: path.resolve(outputFolder, 'images/ec'),
      options: { up: true },
    },
    {
      from: [path.resolve(nodeModules, '@ecl/resources-social-media-icons/dist/sprites/icons-social-media.svg')],
      to: path.resolve(outputFolder, 'images/eu'),
      options: { up: true },
    },
    {
      from: [path.resolve(nodeModules, '@ecl/resources-ec-icons/dist/sprites/icons.svg')],
      to: path.resolve(outputFolder, 'images/ec'),
      options: { up: true },
    },
    {
      from: [path.resolve(nodeModules, '@ecl/resources-eu-icons/dist/sprites/icon.svg')],
      to: path.resolve(outputFolder, 'images/eu'),
      options: { up: true },
    },
  ],
  styles: [
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-icon-ec.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-icon-ec.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-icon-eu.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-icon-eu.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
  ],
};
