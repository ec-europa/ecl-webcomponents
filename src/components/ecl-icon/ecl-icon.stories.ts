import iconPathEc from '@ecl/resources-ec-icons/dist/sprites/icons.svg';
import iconPathEu from '@ecl/resources-eu-icons/dist/sprites/icons.svg';
import iconsAllEc from '@ecl/resources-ec-icons/dist/lists/all.json';

export default {
  title: 'Components/icon',
};

const Template = (args) => `<ecl-icon 
                              icon="${args.icon}"
                              color="${args.color}"
                              size="${args.size}"
                              transform="${args.transform}"
                              path="${args.theme === 'ec' ? iconPathEc : iconPathEu}"
                            ></ecl-icon>`;

export const Icon = Template.bind({});
Icon.storyName = 'default';
Icon.args = {
  theme: 'ec',
  icon: 'close',
  transform: '',
  color: 'default',
  size: '2xl',
};
Icon.argTypes = {
  theme: {
    type: { name: 'select'},
    options: ['ec', 'eu'],
  },
  icon: {
    type: { name: 'select' },
    options: iconsAllEc,
    description: "Add an icon",
  },
  size: {
    type: { name: 'select' },
    options: ['2xs','xs', 's', 'm', 'l', 'xl', '2xl'],
    description: "icon size",
  },
  color: {
    type: { name: 'select' },
    options: ['default', 'primary', 'inverted'],
    description: "icon color",
  },
  transform: {
    type: { name: 'select' },
    description: 'Button icon transform',
    options: [
      'rotate-90',
      'rotate-180',
      'rotate-270',
      'flip-horizontal',
      'flip-vertical',
    ],
  },
};