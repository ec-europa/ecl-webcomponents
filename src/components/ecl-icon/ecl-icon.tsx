import { Component, Prop, h } from '@stencil/core';
import getAssetPath from "../../utils/assetPath";

@Component({
  tag: 'ecl-icon',
  styleUrls: {
    ec: './build/styles/ecl-icon-ec.css',
    eu: './build/styles/ecl-icon-eu.css',
  },
  shadow: false,
  scoped: true,
  assetsDirs: ['build'],
})

export class EclIcon {
  @Prop() styleClass: string = '';
  @Prop() icon: string = '';
  @Prop() size: string = 'xs';
  @Prop() color: string = '';
  @Prop() titleTag: string = '';
  @Prop({ mutable: true }) path: string;
  @Prop() transform: string = '';
  @Prop({ mutable: true }) theme: string;
  @Prop() sprite: string = '';

  componentWillLoad() {
    this.theme = document.documentElement.getAttribute('data-ecl-theme') ?? (this.theme || 'ec');
    
    if (!this.sprite) {
      this.path = getAssetPath(`./build/images/${this.theme}/icons.svg`);
    } else {
      this.path = getAssetPath(`./build/images/${this.theme}/${this.sprite}.svg`);
    }
  }

  getClass(): string {
    const styleClasses = [
      `ecl-icon`,
      `ecl-icon--${this.size}`,
      this.styleClass,
    ];
    if (this.color) {
      styleClasses.push(
        `ecl-icon--${this.color}`
      );
    }
    if (this.transform) {
      styleClasses.push(
        `ecl-icon--${this.transform}`
      );
    }

    return styleClasses.join(' ');
  }

  render() {
    return (
      <svg class={this.getClass()}>
      { this.titleTag ? <title>{this.titleTag}</title> : '' } 
        <use xlinkHref={`${this.path}#${this.icon}`}></use>
      </svg>
    )
  }
}