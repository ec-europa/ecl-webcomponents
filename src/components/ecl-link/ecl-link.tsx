import { Component, h, Prop, Element} from '@stencil/core';

@Component({
  tag: 'ecl-link',
  styleUrls: {
    ec: './build/styles/ecl-link-ec.css',
    eu: './build/styles/ecl-link-eu.css',
  },
  shadow: false,
  scoped: false,
  assetsDirs: ['build'],
})

export class EclLink {
  @Element() el: HTMLElement;
  @Prop() theme: string = 'ec';
  @Prop() path: string;
  @Prop() styleClass: string;
  @Prop() variant: string = 'default';
  @Prop() titleAttr: string;
  @Prop() external: boolean = false;

  getClass(): string {
    const styleClasses = [
      `ecl-link`,
      `ecl-link--${this.variant}`,
      this.styleClass
    ];

    if (this.el.querySelector('ecl-icon')) {
      styleClasses.push('ecl-link--icon');
      if (this.el.querySelector('ecl-icon').getAttribute('slot') === 'icon-after') {
        styleClasses.push('ecl-link--icon-after');
      } else {
        styleClasses.push('ecl-link--icon-before');
      }
    }

    return styleClasses.join(' ');
  }

  componentDidRender() {
    if (this.el.querySelector('ecl-icon')) {
      this.el.querySelector('ecl-icon svg').classList.add('ecl-link__icon');
      if (this.el.querySelector('ecl-icon').getAttribute('slot') === 'icon-after') {
        this.el.querySelector('ecl-icon svg').classList.add('ecl-link__icon--after');
      } else {
        this.el.querySelector('ecl-icon svg').classList.add('ecl-link__icon--before');
      }
    }
  }

  getExternal() {
    return (
      <ecl-icon 
        icon="external"
        style-class={`ecl-link__icon ecl-link__icon--after sc-ecl-link-${this.theme}`}
        size = "2xs"
      >
      </ecl-icon>
    )  
  }

  render() {
    return (
      <a
        class={this.getClass()}
        href={this.path}
        title={this.titleAttr}
      >
        <slot name="icon-before"></slot>
        <slot></slot>
        <slot name="icon-after">{ this.external ? this.getExternal() : '' }</slot>
      </a>
    );
  }
}
