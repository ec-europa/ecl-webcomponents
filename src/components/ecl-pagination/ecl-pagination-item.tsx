import { Component, Prop, h } from '@stencil/core';

@Component({
  tag: 'ecl-pagination-item',
  scoped: false,
  shadow: false,
})

export class EclPaginationItem {
  @Prop() styleClass: string = '';
  @Prop() theme: string = 'ec';
  @Prop() path: string;
  @Prop() ariaLabel: string;
  @Prop() current: boolean;
  @Prop() previous: boolean;
  @Prop() next: boolean;

  getClass(): string {
    const styleClasses = [
      `ecl-pagination__item`,
      `sc-ecl-pagination-${this.theme}`,
      this.styleClass
    ]

    if (this.current) {
      styleClasses.push('ecl-pagination__item--current')
    }
    if (this.previous) {
      styleClasses.push('ecl-pagination__item--previous');
    }
    if (this.next) {
      styleClasses.push('ecl-pagination__item--next');
    }

    return styleClasses.join(' ');
  }

  render() {
    return (
      <li class={this.getClass()}>
      { !this.current ?
        <ecl-link
          theme={this.theme}
          variant="standalone"
          path={this.path}
          aria-label={this.ariaLabel}
          styleClass={`sc-ecl-pagination-${this.theme}`}
        >
          <slot></slot>
        { this.previous || this.next ? 
          <ecl-icon
            icon="corner-arrow"
            size="xs"
            theme={this.theme}
            transform={this.previous ? 'rotate-270' : 'rotate-90'}
            slot={this.previous ? 'icon-before' : 'icon-after'}
            style-class={`sc-ecl-pagination-${this.theme}`}
          ></ecl-icon> : ''
        }
        </ecl-link> : ''
      }
      { this.current ?
        <span class={`ecl-pagination__text ecl-pagination__text--summary sc-ecl-pagination-${this.theme}`}>
          <slot></slot>
        </span> : ''
      }
      {this.current ? 
        <span class={`ecl-pagination__text ecl-pagination__text--full sc-ecl-pagination-${this.theme}`}>
          {this.ariaLabel}
        </span> : ''
      }
      </li>
    )
  };
}

