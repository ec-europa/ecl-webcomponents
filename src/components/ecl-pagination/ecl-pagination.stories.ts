export default {
  title: 'Components/Navigation/pagination',
};

const Template = args =>
  `<ecl-pagination
    theme="${args.theme}"
  >
    <ecl-pagination-item
      path="/example"
      theme="${args.theme}"
      aria-label="Previous page"
      previous
    >
      Previous
    </ecl-pagination-item>
    <ecl-pagination-item
      theme="${args.theme}"
      path="/example"
      aria-label="Page 24"
     >
      24
     </ecl-pagination-item>
    <ecl-pagination-item
      path="/example"
      theme="${args.theme}"
      aria-label="Page 25"
    >
      25
    </ecl-pagination-item>
    <ecl-pagination-item
      path="/example"
      theme="${args.theme}"
      aria-label="Page 26"
      current
    >
      26
    </ecl-pagination-item>
    <ecl-pagination-item
      path="/example"
      theme="${args.theme}"
      aria-label="Page 27"
    >
      27
    </ecl-pagination-item>
    <ecl-pagination-item
      current-page
      theme="${args.theme}"
      aria-label="Page 28"
    >
      28
    </ecl-pagination-item>
    <ecl-pagination-item
      theme="${args.theme}"
      next
      aria-label="Next page"
    >
      Next
    </ecl-pagination-item>
  </ecl-pagination>`;

export const Pagination = Template.bind({});
Pagination.storyName = 'default';
