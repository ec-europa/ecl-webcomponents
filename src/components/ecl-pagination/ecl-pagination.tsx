import { Component, Prop, h, Element } from '@stencil/core';

@Component({
  tag: 'ecl-pagination',
  styleUrls: {
    ec: './build/styles/ecl-pagination-ec.css',
    eu: './build/styles/ecl-pagination-eu.css',
  },
  shadow: false,
  scoped: true,
})

export class EclPagination {
  @Element() el: HTMLElement;
  @Prop() styleClass: string = '';
  @Prop() theme: string = 'ec';

  getClass(): string {
    const styleClasses = [
      `ecl-pagination`,
      this.styleClass
    ]

    return styleClasses.join(' ');
  }

  componentDidRender() {
    const items = this.el.querySelectorAll('.ecl-pagination__item');
    if (items) {
      this.el.querySelector('.ecl-pagination__list').innerHTML = '';
      this.el.querySelector('.ecl-pagination__list').append(...items);
    }
  };

  render() {
    return (
      <div class={this.getClass()}>
        <ul class="ecl-pagination__list">
          <slot></slot>
        </ul>
      </div>
    )
  };
}

