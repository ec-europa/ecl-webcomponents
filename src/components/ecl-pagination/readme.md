# ecl-pagination



<!-- Auto Generated Below -->


## Properties

| Property     | Attribute     | Description | Type      | Default     |
| ------------ | ------------- | ----------- | --------- | ----------- |
| `ariaLabel`  | `aria-label`  |             | `string`  | `undefined` |
| `current`    | `current`     |             | `boolean` | `undefined` |
| `next`       | `next`        |             | `boolean` | `undefined` |
| `path`       | `path`        |             | `string`  | `undefined` |
| `previous`   | `previous`    |             | `boolean` | `undefined` |
| `styleClass` | `style-class` |             | `string`  | `''`        |
| `theme`      | `theme`       |             | `string`  | `'ec'`      |


## Dependencies

### Depends on

- [ecl-link](../ecl-link)
- [ecl-icon](../ecl-icon)

### Graph
```mermaid
graph TD;
  ecl-pagination-item --> ecl-link
  ecl-pagination-item --> ecl-icon
  ecl-link --> ecl-icon
  style ecl-pagination-item fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
