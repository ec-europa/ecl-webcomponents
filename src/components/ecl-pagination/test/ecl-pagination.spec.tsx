import { newSpecPage } from '@stencil/core/testing';
import { EclPagination } from '../ecl-pagination';
import { EclPaginationItem } from '../ecl-pagination-item';
import { EclLink } from '../../ecl-link/ecl-link';
import { EclIcon } from '../../ecl-icon/ecl-icon';

describe('ecl-pagination', () => {
  it('renders', async () => {
    const page = await newSpecPage({
      components: [EclPagination, EclPaginationItem, EclLink, EclIcon],
      html: `<ecl-pagination
        theme="ec"
      >
        <ecl-pagination-item
          path="/example.html"
          theme="ec"
          aria-label="Previous page"
          previous
        >
          Previous
        </ecl-pagination-item>
        <ecl-pagination-item
          theme="ec"
          path="/example.html"
          aria-label="Page 24"
         >
          24
         </ecl-pagination-item>
        <ecl-pagination-item
          path="/example.html"
          theme="ec"
          aria-label="Page 25"
        >
          25
        </ecl-pagination-item>
        <ecl-pagination-item
          path="/example.html"
          theme="ec"
          aria-label="Page 26"
          current
        >
          26
        </ecl-pagination-item>
        <ecl-pagination-item
          path="/example.html"
          theme="ec"
          aria-label="Page 27"
        >
          27
        </ecl-pagination-item>
        <ecl-pagination-item
          current-page
          theme="ec"
          aria-label="Page 28"
        >
          28
        </ecl-pagination-item>
        <ecl-pagination-item
          theme="ec"
          next
          aria-label="Next page"
        >
          Next
        </ecl-pagination-item>
      </ecl-pagination>`,
    });
    expect(page.root).toMatchSnapshot();
  });
});
