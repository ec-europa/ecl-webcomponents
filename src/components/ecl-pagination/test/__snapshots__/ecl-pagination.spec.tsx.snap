// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ecl-pagination renders 1`] = `
<ecl-pagination theme="ec">
  <!---->
  <div class="ecl-pagination">
    <ul class="ecl-pagination__list">
      <li class="ecl-pagination__item ecl-pagination__item--previous sc-ecl-pagination-ec">
        <ecl-link aria-label="Previous page">
          <!---->
          <a aria-label="Previous page" class="ecl-link ecl-link--icon ecl-link--icon-before ecl-link--standalone sc-ecl-pagination-ec" href="/example.html">
            <ecl-icon slot="icon-before" style-class="sc-ecl-pagination-ec">
              <svg class="ecl-icon ecl-icon--rotate-270 ecl-icon--xs ecl-link__icon ecl-link__icon--before sc-ecl-pagination-ec">
                <use xlink:href="/build/images/ec/icons.svg#corner-arrow"></use>
              </svg>
            </ecl-icon>
            Previous
            <slot-fb name="icon-after"></slot-fb>
          </a>
        </ecl-link>
      </li>
      <li class="ecl-pagination__item sc-ecl-pagination-ec">
        <ecl-link aria-label="Page 24">
          <!---->
          <a aria-label="Page 24" class="ecl-link ecl-link--standalone sc-ecl-pagination-ec" href="/example.html">
            24
            <slot-fb name="icon-after"></slot-fb>
          </a>
        </ecl-link>
      </li>
      <li class="ecl-pagination__item sc-ecl-pagination-ec">
        <ecl-link aria-label="Page 25">
          <!---->
          <a aria-label="Page 25" class="ecl-link ecl-link--standalone sc-ecl-pagination-ec" href="/example.html">
            25
            <slot-fb name="icon-after"></slot-fb>
          </a>
        </ecl-link>
      </li>
      <li class="ecl-pagination__item ecl-pagination__item--current sc-ecl-pagination-ec">
        <span class="ecl-pagination__text ecl-pagination__text--summary sc-ecl-pagination-ec">
          26
        </span>
        <span class="ecl-pagination__text ecl-pagination__text--full sc-ecl-pagination-ec">
          Page 26
        </span>
      </li>
      <li class="ecl-pagination__item sc-ecl-pagination-ec">
        <ecl-link aria-label="Page 27">
          <!---->
          <a aria-label="Page 27" class="ecl-link ecl-link--standalone sc-ecl-pagination-ec" href="/example.html">
            27
            <slot-fb name="icon-after"></slot-fb>
          </a>
        </ecl-link>
      </li>
      <li class="ecl-pagination__item sc-ecl-pagination-ec">
        <ecl-link aria-label="Page 28">
          <!---->
          <a aria-label="Page 28" class="ecl-link ecl-link--standalone sc-ecl-pagination-ec">
            28
            <slot-fb name="icon-after"></slot-fb>
          </a>
        </ecl-link>
      </li>
      <li class="ecl-pagination__item ecl-pagination__item--next sc-ecl-pagination-ec">
        <ecl-link aria-label="Next page">
          <!---->
          <a aria-label="Next page" class="ecl-link ecl-link--icon ecl-link--icon-after ecl-link--standalone sc-ecl-pagination-ec">
            Next
            <slot-fb hidden="" name="icon-after"></slot-fb>
            <ecl-icon slot="icon-after" style-class="sc-ecl-pagination-ec">
              <svg class="ecl-icon ecl-icon--rotate-90 ecl-icon--xs ecl-link__icon ecl-link__icon--after sc-ecl-pagination-ec">
                <use xlink:href="/build/images/ec/icons.svg#corner-arrow"></use>
              </svg>
            </ecl-icon>
          </a>
        </ecl-link>
      </li>
    </ul>
  </div>
</ecl-pagination>
`;
